### DISCUSSION TITLE
base.props ResourceDictionary source

### DISCUSSION THREAD

**Discussion by limeniye** [unoplatform/uno#11934] [Q&A] [answered]
I have base.props file:
```XML
<?xml version="1.0" encoding="utf-8"?>
<Project>
	<ItemGroup>
		<None Include="$(MSBuildThisFileDirectory)AppHead.xaml" />
		
		<ApplicationDefinition
			Include="$(MSBuildThisFileDirectory)AppHead.xaml"
			SubType="Designer"
			XamlRuntime="WinUI"
			Generator="MSBuild:Compile"
			Link="AppHead.xaml" />
		<Compile
			Include="$(MSBuildThisFileDirectory)AppHead.xaml.cs"
			XamlRuntime="WinUI"
			DependentUpon="AppHead.xaml"
			Link="AppHead.xaml.cs" />	
	</ItemGroup>
</Project>
```

And also I have AppHead.xaml file such as this example:
https://github.com/unoplatform/Uno.Samples/blob/master/UI/MvuxHowTos/SliderApp/SliderApp.Base/AppHead.xaml

```XML
<local:App x:Class="Ligric.UI.AppHead" xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
           xmlns:local="using:Ligric.UI">

    <local:App.Resources>
		<ResourceDictionary>
			<ResourceDictionary.MergedDictionaries>
				<ResourceDictionary Source="ms-appx:///Ligric.UI/AppResources.xaml" />
			</ResourceDictionary.MergedDictionaries>
		</ResourceDictionary>
	</local:App.Resources>
	
</local:App>
```

I'am tying to add a style reference to my AppResources:
```XML
<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">

    <ResourceDictionary.MergedDictionaries>
        <ResourceDictionary Source="ms-appx:///Styles/test.xaml"/>
    </ResourceDictionary.MergedDictionaries>
</ResourceDictionary>
```
My test.xaml (just an empty file for test):
```XML
<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">

</ResourceDictionary>
```

But I have following result:
![зображення](https://user-images.githubusercontent.com/69314237/230747996-47e30e20-989e-4b6e-be18-882fa4b35aca.png)

The message:
```
Failed to assign to property 'Microsoft.UI.Xaml.ResourceDictionary.Source' because the type 'Windows.Foundation.String' cannot be assigned to the type 'Windows.Foundation.Uri'. [Line: 7 Position: 25]
```

If I'll comment this line  <!--ResourceDictionary Source="ms-appx:///Styles/test.xaml"--/> the issue is disappearing.

The test repository: https://github.com/limeniye/BasePropsWithStyleResourceDictionary

**Comment by jeromelaban** [comment]
Thanks for the report. You'll need to explicitly specify the assembly name when referencing a ResourceDictionary in `AppResources`. See [this documentation](https://platform.uno/docs/articles/guides/how-to-create-control-libraries.html#moving-the-control-style-in-a-separate-resource-dictionary) on how to do it.

**Reply by limeniye** [reply]
I've been tried it already.
[For example:](https://github.com/limeniye/BasePropsWithStyleResourceDictionary/commit/2bc4d7c216b58749284fcd61b7ac8ee1d5b7af91)
```XML
<ResourceDictionary Source="ms-appx:///Ligric.UI/Styles/test.xaml"/>
```

But I have the same issue.

```
Failed to assign to property 'Microsoft.UI.Xaml.ResourceDictionary.Source' because the type 'Windows.Foundation.String' cannot be assigned to the type 'Windows.Foundation.Uri'. [Line: 7 Position: 25]
```

![зображення](https://user-images.githubusercontent.com/69314237/231214062-16ca01e6-cec7-42b5-98cb-9f3f380a4d3d.png)

**Reply by jeromelaban** [reply]
Can you also validate that the XAML file has the "Page" build action? You may also try to see if the `test.xaml` file is in the app package folder.

**Reply by limeniye** [reply]
I think I miss something, but lets check:

1. My test.xaml file inside the Ligric.UI.Styles
![зображення](https://user-images.githubusercontent.com/69314237/231223375-cc91afb7-36eb-456a-b57c-5dc35807e9dc.png)

2. AppResources.xaml file inside the Ligric.UI with test.xaml Source reference:
![зображення](https://user-images.githubusercontent.com/69314237/231224016-d03b68a5-90aa-4020-85a9-6f74e0bf157f.png)

3. AppResources.xaml has a build action "Page"
![зображення](https://user-images.githubusercontent.com/69314237/231227106-2ee38613-f291-4860-849d-084dc08863ff.png)

4. AppHead inside the Ligric.UI.Windows
![зображення](https://user-images.githubusercontent.com/69314237/231224682-7088f6ea-6126-4341-8daf-cc0d587b82d8.png)

4. I don't have any connect pages and currently my Window application running without window
```
<EnableDefaultPageItems>false</EnableDefaultPageItems>
```
That means that I don't use this styles I'am just trying to initialize tham.

**Reply by limeniye** [reply]
Topic is actual

**Reply by limeniye** [reply, marked as answer]
@jeromelaban , thanks. Yes, its need to set Page build but I did it only for AppResources. Thats was my mistake.
